apps/els_lsp/src/els_rename_provider.erl

```diff
@@ -222,18 +222,29 @@ new_name(_, NewName) ->
 
 -spec function_clause_range(poi_range(), els_dt_document:item()) -> poi_range().
 function_clause_range(VarRange, Document) ->
-  FunPOIs = els_poi:sort(els_dt_document:pois(Document, [function_clause])),
-  %% Find beginning of first function clause before VarRange
-  From = case [R || #{range := R} <- FunPOIs, els_range:compare(R, VarRange)] of
-           []        -> {0, 0}; % Beginning of document
-           FunRanges -> maps:get(from, lists:last(FunRanges))
-         end,
-  %% Find beginning of first function clause after VarRange
-  To = case [R || #{range := R} <- FunPOIs, els_range:compare(VarRange, R)] of
-        []                 -> {999999999, 999999999}; % End of document
-        [#{from := End}|_] -> End
-       end,
-  #{from => From, to => To}.
+  POIs = els_poi:sort(els_dt_document:pois(Document, [ function_clause
+                                                     , spec
+                                                     ])),
```

Comment:
This should probably consider all possible top-level forms. I haven't checked, but I'd imagine similar issue will happen with things like:
```
foo(A) -> A.
-define(?MACRO(A), A).
```
and other top-level items (like `-callback` and `-type`).

TBH, I'm not sure why the logic tries to look for the beginning point of the next form. Wouldn't it be enough to look for the end point of the current clause?

Reply:
If I remember correctly, it's to ensure renaming works even when the current clause is not fully parsable (still being written).

Reply:
Good points, I will add support for other top-level forms as well. 
The reason for the currently implemented logic is that function_clause ranges don't contain the beginning and the end of the body of the clause, but rather just the function clause head.